app/upgrades/v20/upgrades.go

```diff
@@ -134,133 +136,512 @@ func InitializeLastProviderConsensusValidatorSet(
+			InitializationParameters: &initParams,
+			PowerShapingParameters:   &powerShapingParams,
+		}
+		resp, err := msgServer.CreateConsumer(ctx, &msgCreateConsumer)
+		if err != nil {
+			return err
+		}
+		ctx.Logger().Info(
+			fmt.Sprintf(
+				"Created consumer with ID(%s), chainID(%s), and spawnTime(%s) from proposal with ID(%d)",
+				resp.ConsumerId, msg.ChainId, initParams.SpawnTime.String(), proposal.Id,
+			),
+		)
+	} else {
+		// ConsumerAdditionProposal that was submitted, but not yet passed
+
+		// first, create a new consumer chain to get a consumer ID
+		metadata, err := metadataFromCAP(msg)
+		if err != nil {
+			return err
+		}
+		msgCreateConsumer := providertypes.MsgCreateConsumer{
+			Signer:                   govKeeper.GetAuthority(),
+			ChainId:                  msg.ChainId,
+			Metadata:                 metadata,
+			InitializationParameters: nil, // to be added to MsgUpdateConsumer
+			PowerShapingParameters:   nil, // to be added to MsgUpdateConsumer
+		}
+		resp, err := msgServer.CreateConsumer(ctx, &msgCreateConsumer)
+		if err != nil {
+			return err
+		}
+		ctx.Logger().Info(
+			fmt.Sprintf(
+				"Created consumer with ID(%s), chainID(%s), and no spawnTime from proposal with ID(%d)",
+				resp.ConsumerId, msg.ChainId, proposal.Id,
+			),
+		)
+
+		// second, replace the message in the proposal with a MsgUpdateConsumer
+		initParams, err := initParamsFromCAP(msg)
+		if err != nil {
+			return err
+		}
+		powerShapingParams, err := powerShapingParamsFromCAP(msg)
+		if err != nil {
+			return err
+		}
+		msgUpdateConsumer := providertypes.MsgUpdateConsumer{
+			Signer:                   govKeeper.GetAuthority(),
+			ConsumerId:               resp.ConsumerId,
+			Metadata:                 nil,
+			InitializationParameters: &initParams,
+			PowerShapingParameters:   &powerShapingParams,
+		}
+		anyMsg, err := codec.NewAnyWithValue(&msgUpdateConsumer)
+		if err != nil {
+			return err
+		}
+		proposal.Messages[0] = anyMsg
+		// TODO: check if we can call SetProposal within govKeeper.Proposals.Walk
+		if err := govKeeper.SetProposal(ctx, proposal); err != nil {
+			return err
+		}
+		ctx.Logger().Info(
+			fmt.Sprintf(
+				"Replaced proposal with ID(%d) with MsgUpdateConsumer - consumerID(%s), chainID(%s), spawnTime(%s)",
+				proposal.Id, resp.ConsumerId, msg.ChainId, initParams.SpawnTime.String(),
+			),
+		)
+	}
+	return nil
+}
 
-		msgMod, ok := content.(*providertypes.ConsumerModificationProposal)
-		if ok {
-			anyMsg, err := MigrateConsumerModificationProposal(*msgMod, govKeeper.GetAuthority())
-			if err != nil {
+// metadataFromCAP returns ConsumerMetadata from a ConsumerAdditionProposal
+func metadataFromCAP(
+	prop *providertypes.ConsumerAdditionProposal,
+) (providertypes.ConsumerMetadata, error) {
+	metadata := providertypes.ConsumerMetadata{
+		Name:        prop.Title,
+		Description: prop.Description,
+		Metadata:    "",
+	}
+	err := providertypes.ValidateConsumerMetadata(metadata)
```

Comment:
This will always return an error since the `metadata.Metadata` is set to an empty `string`.

Reply:
nice catch